library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -0,0 +1,656 @@
+    jdbcStore.setDatabaseUrl(store.getDatabaseUrl());
+    jdbcStore.setConnectionProperties(store.getConnectionProperties());
+
+    jdbcStore.connect();
+
+    HashMap<Integer, KeyValPair<String, String>> partitionToRangeMap = null;
+    try {
+      partitionToRangeMap = JdbcMetaDataUtility.getPartitionedQueryMap(getPartitionCount(),
+          jdbcStore.getDatabaseDriver(), jdbcStore.getDatabaseUrl(), getTableName(), getKey(),
+          store.getConnectionProperties().getProperty(user), store.getConnectionProperties().getProperty(password),
+          whereCondition, emitColumnList);
+    } catch (SQLException e) {
+      LOG.error("Exception in initializing the partition range", e);
+    }
+
+    KryoCloneUtils<AbstractJdbcPollInputOperator<T>> cloneUtils = KryoCloneUtils.createCloneUtils(this);
+
+    for (int i = 0; i <= getPartitionCount(); i++) {
+      AbstractJdbcPollInputOperator<T> jdbcPoller = null;
+
+      jdbcPoller = cloneUtils.getClone();
+
+      jdbcPoller.setStore(store);
+      jdbcPoller.setKey(getKey());
+      jdbcPoller.setPartitionCount(getPartitionCount());
+      jdbcPoller.setPollInterval(getPollInterval());
+      jdbcPoller.setTableName(getTableName());
+      jdbcPoller.setBatchSize(getBatchSize());
+      jdbcPoller.setEmitColumnList(getEmitColumnList());
+
+      store.connect();
+      //The n given partitions are for range queries and n + 1 partition is for polling query
+      //The upper bound for the n+1 partition is set to null since its a pollable partition
+      if (i < getPartitionCount()) {
+        jdbcPoller.setRangeQueryPair(partitionToRangeMap.get(i));
+        isPollable = false;
+      } else {
+        jdbcPoller.setRangeQueryPair(new KeyValPair<String, String>(partitionToRangeMap.get(i - 1).getValue(), null));
+        isPollable = true;
+      }
+      Partition<AbstractJdbcPollInputOperator<T>> po = new DefaultPartition<AbstractJdbcPollInputOperator<T>>(
+          jdbcPoller);
+      newPartitions.add(po);
+    }
+
+    previousUpperBound = null;
+    return newPartitions;
+  }
+
+  @Override
+  public void partitioned(
+      Map<Integer, com.datatorrent.api.Partitioner.Partition<AbstractJdbcPollInputOperator<T>>> partitions)
+  {
+    //Nothing to implement here
+  }
+
+  /**
+   * This class polls a store that can be queried with a JDBC interface The
+   * preparedStatement is updated as more rows are read
+   */
+  public class DBPoller implements Runnable
+  {
+    @Override
+    public void run()
+    {
+      while (execute) {
+        try {
+          long startTs = System.currentTimeMillis();
+          if ((isPollable && !isPolled) || !isPollable) {
+            pollRecords(ps);
+          }
+          long endTs = System.currentTimeMillis();
+          long ioTime = endTs - startTs;
+          long sleepTime = pollInterval - ioTime;
+          LOG.debug("pollInterval = " + pollInterval + " I/O time = " + ioTime + " sleepTime = " + sleepTime);
```

Comment:
use {} syntax for log message, the string gets evaluated even if log level is not enabled.